packages/augur-sdk/src/state/db/DerivedDB.ts

```diff
@@ -65,15 +67,19 @@ export class DerivedDB extends AbstractDB {
         return val;
         }, idDocuments[0]);
       }));
-      documents = _.sortBy(documents, "_id");
 
-      success = await this.bulkUpsertDocuments(documents[0]._id, documents);
+      success = await this.bulkUpsertUnorderedDocuments(documents);
     }
 
-
     if (success) {
       if (!syncing) {
+        // The mutex behavior here is needed since a derived DB may be responding to multiple updates in parallel
+        while (this.updatingHighestSyncBlock) {
+          await sleep(10);
+        }
```

Comment:
It's possible but unlikely that two threads could get past the while-loop at the same time. You can use the mutex library since it uses a queue: https://github.com/DirtyHairy/async-mutex

Reply:
This turned out not to be a concern due to how javascript implements its pseudo-multithreading.